src/nbl/video/SOpenGLState.h

```diff
@@ -0,0 +1,173 @@
+        } compute;
+    } pipeline;
+
+    struct {
+        core::smart_refctd_ptr<const COpenGLBuffer> buffer;
+    } dispatchIndirect;
+
+    struct {
+        // TODO add viewports to flushing routine
+        struct {
+            GLint x = 0;
+            GLint y = 0;
+            // initial values depend on surface attached to context, lets force always updating this (makes life easier, i dont have to query the extents then)
+            GLsizei width = std::numeric_limits<GLsizei>::max();
+            GLsizei height = std::numeric_limits<GLsizei>::max();
+
+            GLdouble minDepth = 0.0;
+            GLdouble maxDepth = 1.0;
+        } viewport[MAX_VIEWPORT_COUNT]; // 16 is max supported viewport count throughout implementations
+        //in GL it is possible to set polygon mode separately for back- and front-faces, but in VK it's one setting for both
+        GLenum polygonMode = GL_FILL;
+        GLenum faceCullingEnable = 0;
+        GLenum cullFace = GL_BACK;
+        //in VK stencil params (both: stencilOp and stencilFunc) are 2 distinct for back- and front-faces, but in GL it's one for both
+        struct SStencilOp {
+            GLenum sfail = GL_KEEP;
+            GLenum dpfail = GL_KEEP;
+            GLenum dppass = GL_KEEP;
+            bool operator!=(const SStencilOp& rhs) const { return sfail!=rhs.sfail || dpfail!=rhs.dpfail || dppass!=rhs.dppass; }
+        };
+        SStencilOp stencilOp_front, stencilOp_back;
+        struct SStencilFunc {
+            GLenum func = GL_ALWAYS;
+            GLint ref = 0;
+            GLuint mask = ~static_cast<GLuint>(0u);
+            bool operator!=(const SStencilFunc& rhs) const { return func!=rhs.func || ref!=rhs.ref || mask!=rhs.mask; }
+        };
+        SStencilFunc stencilFunc_front, stencilFunc_back;
+        // TODO add stencilWriteMask to flushing routine (glStencilMaskSeparate)
+        GLuint stencilWriteMask_front;
+        GLuint stencilWriteMask_back;
+        GLenum depthFunc = GL_LESS;
+        GLenum frontFace = GL_CCW;
+        GLboolean depthClampEnable = GL_FALSE;
+        GLboolean rasterizerDiscardEnable = GL_FALSE;
+        GLboolean polygonOffsetEnable = GL_FALSE;
+        struct SPolyOffset {
+            GLfloat factor = 0.f;//depthBiasSlopeFactor 
+            GLfloat units = 0.f;//depthBiasConstantFactor 
+            bool operator!=(const SPolyOffset& rhs) const { return factor!=rhs.factor || units!=rhs.units; }
+        } polygonOffset;
+        GLfloat lineWidth = 1.f;
+        GLboolean sampleShadingEnable = GL_FALSE;
+        GLfloat minSampleShading = 0.f;
+        GLboolean sampleMaskEnable = GL_FALSE;
+        GLbitfield sampleMask[2]{~static_cast<GLbitfield>(0), ~static_cast<GLbitfield>(0)};
+        GLboolean sampleAlphaToCoverageEnable = GL_FALSE;
+        GLboolean sampleAlphaToOneEnable = GL_FALSE;
+        GLboolean depthTestEnable = GL_FALSE;
+        GLboolean depthWriteEnable = GL_TRUE;
+        //GLboolean depthBoundsTestEnable;
```

Comment:
?

Reply:
idk, it was commented out in previous api as well

Reply:
make an issue on GH about depthBoundsTest

Reply:
no issue still